File: app/src/main/java/com/finnmglas/launcher/FontAwesome.java
package com.finnmglas.launcher;

import android.content.Context;
import android.graphics.Typeface;
import android.util.AttributeSet;
import android.widget.TextView;

public class FontAwesome extends androidx.appcompat.widget.AppCompatTextView {

    public FontAwesome(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
        init();
    }

    public FontAwesome(Context context, AttributeSet attrs) {
        super(context, attrs);
        init();
    }

    public FontAwesome(Context context) {
        super(context);
        init();
    }

    private void init() {

        //Font name should not contain "/".
        Typeface tf = Typeface.createFromAsset(getContext().getAssets(),
                "fonts/fa-solid-900.ttf");
        setTypeface(tf);
    }

}

File: app/src/main/java/com/finnmglas/launcher/FontAwesome.kt
package com.finnmglas.launcher // replace with your package

// On GitHub: https://github.com/finnmglas/fontawesome-android

import android.content.Context
import android.graphics.Typeface
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatTextView


class FontAwesomeSolid : AppCompatTextView {
    constructor(
        context: Context?,
        attrs: AttributeSet?,
        defStyle: Int
    ) : super(context, attrs, defStyle) {
        init()
    }

    constructor(context: Context?, attrs: AttributeSet?) : super(
        context,
        attrs
    ) {
        init()
    }

    constructor(context: Context?) : super(context) {
        init()
    }

    private fun init() {
        typeface = Typeface.createFromAsset(
            context.assets,
            "fontawesome/fa-solid-900.ttf"
        )
    }
}

class FontAwesomeRegular : AppCompatTextView {
    constructor(
        context: Context?,
        attrs: AttributeSet?,
        defStyle: Int
    ) : super(context, attrs, defStyle) {
        init()
    }

    constructor(context: Context?, attrs: AttributeSet?) : super(
        context,
        attrs
    ) {
        init()
    }

    constructor(context: Context?) : super(context) {
        init()
    }

    private fun init() {
        typeface = Typeface.createFromAsset(
            context.assets,
            "fontawesome/fa-regular-400.ttf"
        )
    }
}

class FontAwesomeBrand : AppCompatTextView {
    constructor(
        context: Context?,
        attrs: AttributeSet?,
        defStyle: Int
    ) : super(context, attrs, defStyle) {
        init()
    }

    constructor(context: Context?, attrs: AttributeSet?) : super(
        context,
        attrs
    ) {
        init()
    }

    constructor(context: Context?) : super(context) {
        init()
    }

    private fun init() {
        typeface = Typeface.createFromAsset(
            context.assets,
            "fontawesome/fa-brands-400.ttf"
        )
    }
}
